pkg/distributor/distributor.go

```diff
@@ -284,18 +292,34 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 	var lastPartialErr error
 	removeReplica := false
 
+	numSamples := 0
+	for _, ts := range req.Timeseries {
+		numSamples += len(ts.Samples)
+	}
+	receivedSamples.WithLabelValues(userID).Add(float64(numSamples))
```

Comment:
We use this metric for billing.  By moving it here, customers will get charged for samples which are rejected.

Reply:
Given that, it looks like the whole ha tracking block might need moving below the validation block.

Reply:
The idea was to have a metric to compare total samples in vs # of samples deduped. I can add a new metric instead of using `receivedSamples`. 

Why would we need to move the HA tracking block below validation? For the deduped samples counting, so we only count samples if they user is valid/not above their limit? We'd still want the current block to know if we want to remove the replica label prior to iterating through each series in `req.Timeseries`.